utils/githooks/find_base.sh

```diff
@@ -38,7 +38,7 @@ else
     TARGET="$ORIGIN/master"
     min_diff=-1
     for base in "${all_bases[@]}"; do
-        git rev-parse --verify "$base" 2&>/dev/null || continue
+        _=$(git rev-parse --verify "$ORIGIN/$base") || continue
```

Comment:
What exactly is the issue here that you are junking the output into `$_`?

Reply:
Definitely open to bash suggestions. We don't get along :)

Reply:
So the relationship between this and a caller is that if a caller is using `set -e` and does not check the return of this script'S execution in any way, any `!0` exit of this script will cause the caller to take it's `ERROR` handling path (i.e. exit immediately or call the `ERROR` `trap` if one is set).

I suspect what you are looking for is this:

```suggestion
        git rev-parse --verify "$ORIGIN/$base" &> /dev/null || continue
```

to discard both `stdout` and `stderr`.

Reply:
Ah, yes! Using `2&>` - not correct
```
$ git rev-parse --verify origin/foo 2&> /dev/null || echo hello
hello
$ git rev-parse --verify origin/master 2&> /dev/null || echo hello
hello
```
using `&>` - correct
```
$ git rev-parse --verify origin/foo &> /dev/null || echo hello
hello
$ git rev-parse --verify origin/master &> /dev/null || echo hello
# no output
```
Thanks!